File: cli/commands/status.ts
/**
 * Status command - Check daemon status
 */
import { getDaemonStatus } from "../daemon-manager.js";

export async function statusDaemon(): Promise<void> {
  const status = await getDaemonStatus();
  
  if (!status.running) {
    console.log("Daemon is not running");
    console.log("Run 'console start' to start the daemon");
    process.exit(0);
  }

  console.log("Daemon is running");
  console.log(`PID: ${status.pid}`);
  console.log(`Uptime: ${status.uptime || "unknown"}`);
  console.log(`Server: http://${status.host}:${status.port}`);
  console.log(`Logs: ~/.console/logs/daemon.log`);
  console.log(`Run 'console logs' to view logs`);
  console.log(`Run 'console stop' to stop the daemon`);
}

File: apps/server-go/internal/cli/daemon/status.go
// Daemon status: running/pid/uptime/port/host/mode. Port of getDaemonStatus
// in apps/cli/daemon-manager.ts, including the dual-directory fallback so
// stop/status work regardless of which storage mode the running daemon used.
package daemon

import (
	"os/exec"
	"strconv"
	"strings"
)

// Status is the point-in-time daemon status.
type Status struct {
	Running bool
	Pid     int
	Uptime  string
	Port    string
	Host    string
	Mode    Mode
}

// GetStatus checks ConsoleDir first, then AlternateConsoleDir, clearing any
// stale PID file it finds along the way.
func GetStatus() Status {
	for _, dir := range []string{ConsoleDir(), AlternateConsoleDir()} {
		pid := ReadPidFile(dir)
		if pid == 0 {
			continue
		}
		if !IsProcessRunning(pid) {
			RemovePidFile(dir)
			continue
		}
		return buildRunningStatus(pid, dir)
	}
	return Status{Running: false}
}

func buildRunningStatus(pid int, dir string) Status {
	uptime := ""
	out, err := exec.Command("ps", "-p", strconv.Itoa(pid), "-o", "etime=").Output()
	if err != nil {
		// Process might have died between the running-check and here.
		return Status{Running: false}
	}
	uptime = strings.TrimSpace(string(out))

	cfg := LoadConfig(dir)
	mode := ModeProduction
	if dir == ConsoleDir() {
		mode = ResolveMode()
	} else {
		mode = alternateMode()
	}
	return Status{
		Running: true, Pid: pid, Uptime: uptime,
		Port: cfg.Port, Host: cfg.Host, Mode: mode,
	}
}
